### ISSUE TITLE
version 2.x  converts regular expressions to objects when cloning

### ISSUE THREAD

**Issue by cressie176** [swordev/merge#41]
Thanks for sharing merge. When upgrading from v1 to v2 I found the following bug. Bit confused as what could have caused it as the clone code doesn't seem to have changed greatly (other than being converted to TypeScript)

```js
const merge = require('merge');
const result = merge.recursive(true, { regex: /.*/ } , {});
console.log(result);
```
### v1.x
```
{ regex: /.*/ }
```
### v2.x
```
{ regex: {} }
```

**Comment by cressie176** [comment]
https://github.com/yeikos/js.merge/blob/master/src/index.ts#L55-L57 is the problem.

**Comment by dmellstrom** [comment]
@cressie176
I realize this is slightly off-topic, but are you aware of any other breaking changes in 2.x? I'm thinking of updating this dep in a few places. Nothing was documented that I can see, and it doesn't appear there were any **intentional** changes to the API with the Typescript migration, but I thought I'd ask around to be safe.

@yeikos care to weigh in?

**Comment by tferrin** [comment]
This bug recently bit me and required more than a day of sleuthing before I figured it out. Hard to believe it is still an open issue more than 18 months after getting reported, especially since [@cressie176](url) pointed out where the problem lies.

**Comment by juanrgm** [comment]
Closed via e9bea183aa374810ab2262639e303eb0500e6790.
